controllers/istio/istio_controller_test.go

```diff
@@ -1328,6 +1329,182 @@ func TestMergeOverwrite(t *testing.T) {
 	}
 }
 
+func TestApplyImageDigests(t *testing.T) {
+	testCases := []struct {
+		name         string
+		config       common.OperatorConfig
+		inputIstio   *v1alpha1.Istio
+		inputValues  helm.HelmValues
+		expectValues helm.HelmValues
+	}{
+		{
+			name: "no-user-values",
+			config: common.OperatorConfig{
```

Comment:
can you add a test with no (or empty) `config`?

Reply:
done!